rackcity/views/it_model_views.py

```diff
@@ -50,25 +54,46 @@ def model_add(request):
     Add a new model
     """
     data = JSONParser().parse(request)
-    failure_message = ""
     if 'id' in data:
-        failure_message = failure_message + "Don't include id when adding a model. "
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.CREATE_ERROR.value + GenericFailure.UNKNOWN.value,
+                "errors": "Don't include 'id' when creating a model"
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
     serializer = ITModelSerializer(data=data)
     if not serializer.is_valid(raise_exception=False):
-        failure_message = failure_message + str(serializer.errors)
-    if failure_message == "":
-        try:
-            new_model = serializer.save()
-            log_action(request.user, new_model, Action.CREATE)
-            return HttpResponse(status=HTTPStatus.CREATED)
-        except Exception as error:
-            failure_message = failure_message + str(error)
-
-    failure_message = "Request was invalid. " + failure_message
-    return JsonResponse({
-        "failure_message": failure_message
-    },
-        status=HTTPStatus.NOT_ACCEPTABLE
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.CREATE_ERROR.value +
+                    GenericFailure.INVALID_DATA.value,
+                "errors": str(serializer.errors)
```

Comment:
This means we lose some detail (uniqueness validation, invalid value validation, etc) because we're not returning the serializer's json gobbly gook answer. To get the specifics of what went wrong in the serialization, we could parse the serializer errors, but saving that improvement for another PR / time permitting